backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/VersionInputHandler.java

```diff
@@ -0,0 +1,245 @@
+                  + entityIdReferenceColumn
+                  + ") "
+                  + " VALUES (:repository_id, :commit, :versioning_key, :versioning_location, :entity_type, :versioning_blob_type, :blob_hash, :entityId)";
+          if (versioningEntry.getKeyLocationMapMap().isEmpty()) {
+            handle
+                .createUpdate(queryStr)
+                .bind("repository_id", versioningEntry.getRepositoryId())
+                .bind("commit", versioningEntry.getCommit())
+                .bind("versioning_key", ModelDBConstants.EMPTY_STRING)
+                .bind("versioning_location", ModelDBConstants.EMPTY_STRING)
+                .bind("versioning_blob_type", ModelDBConstants.EMPTY_STRING)
+                .bind("blob_hash", ModelDBConstants.EMPTY_STRING)
+                .bind("entity_type", entity_type)
+                .bind("entityId", entityId)
+                .execute();
+          } else {
+            for (Map.Entry<String, Location> locationEntry :
+                versioningEntry.getKeyLocationMapMap().entrySet()) {
+              String locationKey = String.join("#", locationEntry.getValue().getLocationList());
+              Map.Entry<BlobExpanded, String> blobExpandedWithHashMap =
+                  locationBlobWithHashMap.get(locationKey);
+
+              Blob blob = blobExpandedWithHashMap.getKey().getBlob();
+
+              handle
+                  .createUpdate(queryStr)
+                  .bind("repository_id", versioningEntry.getRepositoryId())
+                  .bind("commit", versioningEntry.getCommit())
+                  .bind("versioning_key", locationEntry.getKey())
+                  .bind(
+                      "versioning_location",
+                      ModelDBUtils.getStringFromProtoObject(locationEntry.getValue()))
+                  .bind("versioning_blob_type", blob.getContentCase().getNumber())
+                  .bind("blob_hash", blobExpandedWithHashMap.getValue())
+                  .bind("entity_type", entity_type)
+                  .bind("entityId", entityId)
+                  .execute();
+
+              if (blob.getContentCase().equals(Blob.ContentCase.CONFIG)) {
+                var configBlobQuery =
+                    "SELECT cb.blob_hash, cb.config_seq_number, cb.hyperparameter_type, hecbs.name, hecbs.int_value, hecbs.float_value, hecbs.string_value "
+                        + " FROM config_blob cb "
+                        + " JOIN hyperparameter_element_config_blob hecbs ON cb.hyperparameter_element_config_blob_hash = hecbs.blob_hash"
+                        + " WHERE cb.blob_hash = :blobHash";
+                handle
+                    .createQuery(configBlobQuery)
+                    .bind("blobHash", blobExpandedWithHashMap.getValue())
+                    .map(
+                        (rs, ctx) -> {
+                          String blobMappingQueryStr =
```

Comment:
This seems to be executed multiple times, so we should use the JDBI batch functionality.

Reply:
Implemented batch for entire class where batch is needed.